addons/point_of_sale/models/pos_order.py

```diff
@@ -456,16 +459,18 @@ def action_invoice(self):
             inv = invoice._convert_to_write({name: invoice[name] for name in invoice._cache})
             new_invoice = Invoice.with_context(local_context).sudo().create(inv)
             message = _("This invoice has been created from the point of sale session: <a href=# data-oe-model=pos.order data-oe-id=%d>%s</a>") % (order.id, order.name)
-            new_invoice.message_post(body=message)            
+            new_invoice.message_post(body=message)
             order.write({'invoice_id': new_invoice.id, 'state': 'invoiced'})
             Invoice += new_invoice
 
             for line in order.lines:
                 self.with_context(local_context)._action_create_invoice_line(line, new_invoice.id)
 
             new_invoice.with_context(local_context).sudo().compute_taxes()
-            order.sudo().signal_workflow('invoice')
-            new_invoice.sudo().signal_workflow('validate')
+            order.sudo().write({'state': 'invoiced'})
+            # this workflow signal didn't exist on account.invoice -> should it have been 'invoice_open' ? (and now method .action_invoice_open())
```

Comment:
Nope, it wasn't us! I can't find any trace of this "validate" wkf signal in the invoice workflow (checked from 6.1 to 9.0). I have no idea what that was supposed to do.

I do agree that the invoice **could** be marked as paid, but it has the disadvantage of validating etc. the invoice, leaving no room for edition later if needed. Not sure it should be done yet, maybe ask @mart-e since he is now the pos-master. I'm not exactly clear on POSi-invoicing flows (i.e. is it possible to print the invoice directly from the pos?), so I don't really have a meaningful opinion here...